File: src/test/java/idlezoo/game/services/ResourcesServiceTest.java
package idlezoo.game.services;


import idlezoo.game.domain.Building;
import idlezoo.game.domain.Perks.Perk;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import static org.junit.jupiter.api.Assertions.*;

@ExtendWith(SpringExtension.class)
@SpringBootTest
class ResourcesServiceTest {

    @Autowired
    private ResourcesService gameResources;

    @Test
    void testAnimals() {
        assertNotNull(gameResources.startingAnimal());
        assertSame(gameResources.startingAnimal(), gameResources.getAnimalsList().get(0));
        Building previous = gameResources.startingAnimal();
        for (Building creature : gameResources.getAnimalsList()) {
            if (creature == previous) {
                continue;
            }
            assertTrue(previous.getBaseCost() < creature.getBaseCost());
            assertTrue(previous.getBaseIncome() < creature.getBaseIncome());
            assertTrue(previous.getBaseUpgrade() < creature.getBaseUpgrade());
            previous = creature;
        }
    }

    @Test
    void testPerks() {
        assertEquals("TODO100", gameResources.getPerkList().get(0).getName());
        assertEquals(1e7, gameResources.getPerkList().get(0).getCost(), 0.0001);

        for (Perk perk : gameResources.getPerkList()) {
            Integer index = gameResources.perkIndex(perk.getName());
            assertEquals(perk, gameResources.perkByIndex(index));
            assertEquals(perk, gameResources.perk(perk.getName()));
        }
    }
}

File: src/test/kotlin/idlezoo/game/services/ResourcesServiceTest.kt
package idlezoo.game.services


import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.NONE
import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = NONE)
class ResourcesServiceTest(@Autowired val gameResources: ResourcesService) {

    @Test
    fun testAnimals() {
        assertNotNull(gameResources.startingAnimal())
        assertSame(gameResources.startingAnimal(), gameResources.animalsList[0])
        var previous = gameResources.startingAnimal()
        for (creature in gameResources.animalsList) {
            if (creature === previous) {
                continue
            }
            assertTrue(previous!!.baseCost < creature.baseCost)
            assertTrue(previous.baseIncome < creature.baseIncome)
            assertTrue(previous.baseUpgrade < creature.baseUpgrade)
            previous = creature
        }
    }

    @Test
    fun testPerks() {
        assertEquals("TODO100", gameResources.perkList[0].name)
        assertEquals(1e7, gameResources.perkList[0].cost, 0.0001)

        for (perk in gameResources.perkList) {
            val index = gameResources.perkIndex(perk.name)
            assertEquals(perk, gameResources.perkByIndex(index!!))
            assertEquals(perk, gameResources.perk(perk.name))
        }
    }
}
